modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/modal_content/DateRangeFilter.tsx

```diff
@@ -5,102 +5,243 @@
 
 import ClayDatePicker from '@clayui/date-picker';
 import ClayForm from '@clayui/form';
+import ClayModal from '@clayui/modal';
 import classNames from 'classnames';
 import {format, getYear, isBefore, isEqual} from 'date-fns';
-import React, {useEffect} from 'react';
+import React, {useState} from 'react';
+
+import FilterModalConfiguration from '../../../components/FilterModalConfiguration';
+import FilterModalFooter from '../../../components/FilterModalFooter';
+import {IDateFilter, IField, IFilter} from '../../../utils/types';
 
 function Header() {
 	return <>{Liferay.Language.get('new-date-range-filter')}</>;
 }
 
 interface IBodyProps {
-	from: string;
-	isValidDateRange: boolean;
+	closeModal: Function;
+	fieldNames?: string[];
+	fields: IField[];
+	filter?: IFilter;
+	handleSave: Function;
 	namespace: string;
-	onFromChange: (val: string) => void;
-	onToChange: (val: string) => void;
-	onValidDateChange: (val: boolean) => void;
-	to: string;
 }
 
 function Body({
-	from,
-	isValidDateRange,
+	closeModal,
+	fieldNames,
+	fields,
+	filter,
+	handleSave,
 	namespace,
-	onFromChange,
-	onToChange,
-	onValidDateChange,
-	to,
 }: IBodyProps) {
+	const [fieldInUseValidationError, setFieldInUseValidationError] = useState<
+		boolean
+	>(false);
```

Comment:
Can this be a state variable in the common section component?

Reply:
This one looks redundant if validation is moved to `FilterModalConfiguration`, similar consideration for ClientExtensionFilter and SelectionFilter as these state items are also there

Reply:
See previous comment